src/zenml/utils/daemon.py

```diff
@@ -198,7 +199,9 @@ def run_as_daemon(
             pid = os.fork()
             if pid > 0:
                 # this is the process that called `run_as_daemon` so we
-                # simply return so it can keep running
+                # wait for the child process to finish and simply return
+                # so it can keep running
+                os.wait()
```

Comment:
I'll update the comment. As a side note, three different processes are involved here in a parent-child daisy chain. The middle process only lives for a fraction of a second, but we still need to wait for it to finish, otherwise it becomes a zombie that will haunt human kind forever.